tools.py

```diff
@@ -44,20 +44,31 @@ def cut(input_path, output_path, sample_rate, mono, dilate, strip, strip_prefix,
 		signal = audio.read_audio(audio_path, sample_rate, normalize = False, backend = audio_backend)[0] if audio_path != prev_audio_path else signal
 		t['channel'] = 0 if len(signal) == 1 else None if mono else t.get('channel')
 		segment = signal[slice(t['channel'], 1 + t['channel']) if t['channel'] is not None else ..., int(max(t['begin'] - dilate, 0) * sample_rate) : int((t['end'] + dilate) * sample_rate)]
-		segment_path = os.path.join(output_path, os.path.basename(audio_path) + '.{channel}-{begin:.06f}-{end:.06f}.wav'.format(**t))
+		
+		segment_file_name = os.path.basename(audio_path) + '.{channel}-{begin:.06f}-{end:.06f}.wav'.format(**t)
+		sub_path = [str(hash(segment_file_name) % 10), str(hash(segment_file_name) % 1000)] if add_sub_paths else []
```

Comment:
better use some stable, fixed hash implementation like md5 or sha1 (python hash implementation may change when python evolves). example: https://stackoverflow.com/a/5297483/445810

side note: can hash return negative values? If yes, remainder operator may preserve the sign and you'd get minus sign in the directory names